File: cpp/2586. 统计范围内的元音字符串数.go
func vowelStrings(words []string, left int, right int) int {
    vowels := map[byte]struct{}{'a' : {}, 'e' : {}, 'i' : {}, 'o' : {}, 'u' : {}}
    ans := 0
    for _, word := range(words[left : right + 1]) {
        if _, ok1 := vowels[word[0]]; ok1 {
            if _, ok2 := vowels[word[len(word) - 1]]; ok2 {
                ans++
            }
        }
    }
    return ans
}

File: cpp/2586. 统计范围内的元音字符串数.cpp
class Solution {
public:
    bool isVowel(char c) {
        if(c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'){
            return true;
        }
        return false;
    }
    int vowelStrings(vector<string>& words, int left, int right) {
        int ans=0;
        for(int i=left; i<=right; i++){
            string word=words[i];
            if(isVowel(word[0]) && isVowel(word[word.length()-1])){
                ans++;
            }
        }
        return ans;
    }
};